sources/bloodhound/src/bin/kubernetes-checks/checks.rs

```diff
@@ -0,0 +1,724 @@
+                .to_string(),
+            id: "4.2.5".to_string(),
+            level: 1,
+            name: "k8s04020500".to_string(),
+            mode: Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct K8S04020600Checker {}
+
+impl Checker for K8S04020600Checker {
+    fn execute(&self) -> CheckerResult {
+        #[derive(Deserialize)]
+        struct KubeletConfig {
+            #[serde(rename = "makeIPTablesUtilChains")]
+            make_iptables_util_chains: bool,
+        }
+
+        let mut result = CheckerResult::default();
+
+        if let Ok(kubelet_file) = File::open(KUBELET_CONF_FILE) {
+            if let Ok(config) = serde_yaml::from_reader::<_, KubeletConfig>(kubelet_file) {
+                if !config.make_iptables_util_chains {
+                    result.error = "Kubelet makeIPTablesUtilChains is disabled".to_string();
+                    result.status = CheckStatus::FAIL;
+                } else {
+                    result.status = CheckStatus::PASS;
+                }
+            } else {
+                // Normally this value should not be present in the config file, so deserialization is expected to fail.
+                result.status = CheckStatus::PASS;
+            }
+        } else {
+            result.error = format!("unable to read '{}'", KUBELET_CONF_FILE);
+        }
+
+        result
+    }
+
+    fn metadata(&self) -> CheckerMetadata {
+        CheckerMetadata {
+            title: "Ensure that the --make-iptables-util-chains argument is set to true"
+                .to_string(),
+            id: "4.2.6".to_string(),
+            level: 1,
+            name: "k8s04020600".to_string(),
+            mode: Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct K8S04020900Checker {}
+
+impl Checker for K8S04020900Checker {
+    fn execute(&self) -> CheckerResult {
+        #[derive(Deserialize)]
+        struct KubeletConfig {
+            #[serde(rename = "tlsCertFile")]
+            tls_cert_file: String,
+            #[serde(rename = "tlsPrivateKeyFile")]
+            tls_private_key_file: String,
+        }
+
+        let mut result = CheckerResult::default();
+
+        if let Ok(kubelet_file) = File::open(KUBELET_CONF_FILE) {
+            if let Ok(config) = serde_yaml::from_reader::<_, KubeletConfig>(kubelet_file) {
+                if (!config.tls_cert_file.is_empty() && Path::new(&config.tls_cert_file).exists())
+                    && (!config.tls_private_key_file.is_empty()
+                        && Path::new(&config.tls_private_key_file).exists())
+                {
+                    result.status = CheckStatus::PASS;
+                } else {
+                    result.error = "TLS files not set to expected path".to_string();
```

Comment:
Should the expected path be included in this error?

Reply:
It is dependent on what is in the config file, which may not contain anything. So this is a general message to point at where the issue is, and the benchmark contains the steps necessary to manually verify if these values are not found in the config file where they are expected to be.